### PR TITLE
WIP TEST BRANDING

### PR DESCRIPTION
**Pull request by kebeclibre**
OPW 2007312

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by kebeclibre** [comment]
@qsm-odoo 
I am working on the fix, but can you confirm the test is correct ?

File: odoo/addons/base/tests/test_views.py [hunk 1]
```diff
@@ -674,6 +674,45 @@ def test_branding_inherit(self):
...
+        arch = etree.fromstring(arch_string)
+        self.View.distribute_branding(arch)
+
+        for i in [1, 3]:
+            xpath = '/hello[1]/world[%s]' % i
+            [initial] = arch.xpath(xpath)
+            self.assertEqual(
+                xpath,
+                initial.get('data-oe-xpath'),
+                "initial should come from the root view")
```
Comment (qsm-odoo) [inline odoo/addons/base/tests/test_views.py]: I think that this `assert` is testing what it should for the third world (the only really useful assert in the test is that one) but it is not correct to test the first world with it. As the first one comes from the inherit view, I think its oe-xpath is very different (something like `/xpath[1]/world[1]`) ?

**Comment by qsm-odoo** [comment]
(Thanks for working on this BTW :))

**Comment by kebeclibre** [comment]
@qsm-odoo 
I made quite a progress, although I'd need your input at this point because I feel my fix is quite inelegant, but also that I don't have a choice
Tests are extensive mostly because I don't know what cases are tested elsewhere
Also, I think the more extensive they are, the more they speak for themselves

File: odoo/addons/base/models/ir_ui_view.py [hunk 2]
```diff
@@ -646,9 +646,13 @@ def extract(spec):
                     if node.getparent() is None:
                         source = copy.deepcopy(spec[0])
                     else:
+                        replaced_set = False
```
Comment (dmo-odoo) [inline odoo/addons/base/models/ir_ui_view.py]: not a set

File: odoo/addons/base/models/ir_ui_view.py [hunk 3]
```diff
@@ -646,9 +646,13 @@ def extract(spec):
...
                         for child in spec:
                             if child.get('position') == 'move':
                                 child = extract(child)
+                            if self._context.get('inherit_branding') and not replaced_set and child.tag is not etree.Comment:
```
Comment (dmo-odoo) [inline odoo/addons/base/models/ir_ui_view.py]: comment

**Comment by kebeclibre** [comment]
robodoo r+

**Pull request closed without merging**